### PR TITLE
docs: fix br tools compatibility

### PR DESCRIPTION
**Pull request by ran-huang**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [ ] v7.4 (TiDB 7.4 versions)
- [x] v7.3 (TiDB 7.3 versions)
- [x] v7.2 (TiDB 7.2 versions)
- [x] v7.1 (TiDB 7.1 versions)
- [ ] v7.0 (TiDB 7.0 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/15012
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: br/backup-and-restore-overview.md [hunk 1]
```diff
@@ -119,15 +119,15 @@ Backup and restore might go wrong when some TiDB features are enabled or disable
...
 ### Version compatibility
 
-Before performing backup and restore, BR compares and checks the TiDB cluster version with its own. If there is a major-version mismatch, BR prompts a reminder to exit. To forcibly skip the version check, you can set `--check-requirements=false`. Note that skipping the version check might introduce incompatibility.
-
 > **Note:**
 >
-> When performing backup and restore, it is recommended that the BR and TiDB cluster are of the same version.
+> It is recommended to use the BR of the same major version as your TiDB cluster for backup and restoration.
+
+Before performing backup and restore, BR compares the TiDB cluster version with its own and checks their compatibility. If the versions are incompatible, BR reports an error and exits. To forcibly skip the version check, you can set `--check-requirements=false`. Note that skipping the version check might introduce incompatibility in data.
```
Comment (lilin90) [inline br/backup-and-restore-overview.md:127]: It seems that the content in line 130~132 in https://github.com/pingcap/docs-cn/pull/15012/files# is missed. Please check and update. Thanks!

File: br/backup-and-restore-overview.md [hunk 2]
```diff
@@ -119,15 +119,19 @@ Backup and restore might go wrong when some TiDB features are enabled or disable
...
+
+Starting from v7.0.0, TiDB gradually supports performing backup and restore operations through SQL statements. Therefore, it is strongly recommended to use the BR tool of the same major version as the TiDB cluster when backing up and restoring clusters, and avoid performing data backup and restore operations across major versions. This helps ensure smooth execution of restore operations and data consistency.
```
Comment (lilin90) [inline br/backup-and-restore-overview.md:128]:
```suggestion
Starting from v7.0.0, TiDB gradually supports performing backup and restore operations through SQL statements. Therefore, it is strongly recommended to use the BR tool of the same major version as the TiDB cluster when backing up and restoring cluster data, and avoid performing data backup and restore operations across major versions. This helps ensure smooth execution of restore operations and data consistency.
```

**Review by benmaoer — APPROVED** [review]
LGTM

**Review by lilin90 — APPROVED** [review]
/lgtm

**Pull request merged**